src/main/scala/is/hail/methods/PCRelate.scala

```diff
@@ -239,12 +264,16 @@ class PCRelate(maf: Double, blockSize: Int) extends Serializable {
     * {@code pcs} is sample by numPCs
     *
     **/
-  private[methods] def mu(g: IndexedRowMatrix, pcs: DenseMatrix[Double]): M = {
-    val beta = fitBeta(g, pcs, blockSize)
+  private[methods] def mu(blockedG: M, pcs: DenseMatrix[Double]): M = {
```

Comment:
Given how many downstream operations start from mu, some of which are multiplies, I think you should be caching `mu` or even writing and reading `mu` at this stage (it's pretty fast). You don't want to repeat it's computation, and you're cacheing a ton of other block matrices that might push it out if n is huge depending on cluster size.